ui/packages/catalog-ui-search/src/main/webapp/component/query-advanced/query-advanced.view.js

```diff
@@ -25,6 +25,30 @@ const store = require('../../js/store.js')
 const QuerySettingsView = require('../query-settings/query-settings.view.js')
 const user = require('../singletons/user-instance.js')
 
+import query from '../../react-component/utils/query'
+
+const getCacheKey = (attr, obj) =>
+  JSON.stringify(attr) + JSON.stringify(obj.src)
+
+const fetchFacets = memoize(async (attr, { src: srcs }) => {
```

Comment:
How significant is the performance gain for memoizing vs not? Do the amount of network calls stemming from a typical advanced search after adding faceting necessitate this?

Reply:
Because of the way that marionette works with react, the query form re-renders multiple times when you change between attributes that have a diffferent set of comparators (e.g. Float vs String). That makes the network request go out several times and memoize was a way to easily prevent that from happening.